core/unit_test/TestTeamPolicyConstructors.hpp

```diff
@@ -156,4 +169,34 @@ TEST(TEST_CATEGORY, team_policy_runtime_parameters) {
   // clang-format on
 }
 
+// The execution space is defaulted if not given to the constructor.
+TEST(TEST_CATEGORY, team_policy_default_space) {
+  using policy_t = Kokkos::TeamPolicy<TEST_EXECSPACE>;
+
+  policy_t defaulted(42, Kokkos::AUTO);
+
+  ASSERT_EQ(defaulted.space(), TEST_EXECSPACE{});
+}
+
+// The execution space instance can be updated.
+TEST(TEST_CATEGORY, team_policy_impl_set_space) {
+  using policy_t = Kokkos::TeamPolicy<TEST_EXECSPACE>;
+
+  const TEST_EXECSPACE exec{};
+
+#ifdef KOKKOS_ENABLE_OPENMP  // FIXME_OPENMP partition_space
+  if (exec.concurrency() < 2)
+    GTEST_SKIP() << "insufficient number of supported concurrent threads";
+#endif
+
+  const auto [exec_old, exec_new] =
+      Kokkos::Experimental::partition_space(exec, 1, 1);
```

Comment:
I am tempted to say we'd want to check
```suggestion
      ASSERT_NEQ(exec_old, exec_new);
```
but I suppose that is a legal implementation of `Experimental::partition_space()` to return copies of the original exec space instance argument...

Reply:
Yeah, that's why I'm not checking that `policy_new.space()` is different from `exec_old` :wink: